tests/s2n_test.h

```diff
@@ -39,40 +39,63 @@ int test_count;
+ * should happen in main() and start with a BEGIN_TEST() and end with an
+ * END_TEST().
  */
-#define BEGIN_TEST()                                           \
-  do {                                                         \
-    test_count = 0;                                            \
-    EXPECT_SUCCESS_WITHOUT_COUNT(s2n_in_unit_test_set(true));  \
-    S2N_TEST_OPTIONALLY_ENABLE_FIPS_MODE();                    \
-    EXPECT_SUCCESS_WITHOUT_COUNT(s2n_init());                  \
-    fprintf(stdout, "Running %-50s ... ", __FILE__);           \
-  } while(0)
-
-#define END_TEST()   do { \
-                        EXPECT_SUCCESS_WITHOUT_COUNT(s2n_in_unit_test_set(false));      \
-                        EXPECT_SUCCESS_WITHOUT_COUNT(s2n_cleanup());       \
-                        if (isatty(fileno(stdout))) { \
-                            if (test_count) { \
-                                fprintf(stdout, "\033[32;1mPASSED\033[0m %10d tests\n", test_count ); \
-                            }\
-                            else {\
-                                fprintf(stdout, "\033[33;1mSKIPPED\033[0m       ALL tests\n" ); \
-                            }\
-                       } \
-                       else { \
-                            if (test_count) { \
-                                fprintf(stdout, "PASSED %10d tests\n", test_count ); \
-                            }\
-                            else {\
-                                fprintf(stdout, "SKIPPED       ALL tests\n" ); \
-                            }\
-                       } \
-                       return 0;\
-                    } while(0)
+#define BEGIN_TEST()                                                \
+    do {                                                            \
+        test_count = 0;                                             \
+        EXPECT_SUCCESS_WITHOUT_COUNT(s2n_in_unit_test_set(true));   \
+        S2N_TEST_OPTIONALLY_ENABLE_FIPS_MODE();                     \
+        EXPECT_SUCCESS_WITHOUT_COUNT(s2n_init());                   \
+        fprintf(stdout, "Running %-50s ... ", __FILE__);            \
+    } while(0)
+
+#define END_TEST()                                                  \
+    do {                                                            \
+        EXPECT_SUCCESS_WITHOUT_COUNT(s2n_in_unit_test_set(false));  \
+        EXPECT_SUCCESS_WITHOUT_COUNT(s2n_cleanup());                \
+        END_TEST_PRINT()                                            \
+        return 0;                                                   \
+    } while(0)
+
+/* Macros similar to BEGIN_TEST() and END_TEST() but for tests where s2n should
+ * not initialise at the start of the test. Useful for tests that e.g spawn a
+ * number of independent childs at the start of a unit test.
+ */
+#define BEGIN_TEST_NO_INIT()                                        \
+    do {                                                            \
```

Comment:
BEGIN_TEST should probably just call BEGIN_TEST_NO_INIT, and then init. Same for END_TEST. Otherwise, since these is only used in one test, I wouldn't bother with putting them in s2n_test.h.